W-10648006 : Adding additional information to the execute command (#1798)
mulesoft/docs-connectors: mongodb/6.3/modules/ROOT/pages/mongodb-connector-additional-configuration.adoc

= MongoDB Connector Additional Configuration Information - Mule 4

== Additional Results in the Response of the Execute Command Operation

> The *Execute Command* operation is not a paginated operation, as `executeCommand` returns only results with a single invocation.

Reviewer: ```suggestion
The *Execute Command* operation returns one page of results.
```

> If an execution returns a cursor reference, which contains results with additional elements, the transversal of the results of the command execution must be done by invoking an additional `getMore` command.

Reviewer: ```suggestion
If an execution returns a cursor reference, invoke the `getMore` command as needed to traverse subsequent pages of results.
```

Suppose `executeCommand` is invoked with the following command value:

[source,dataweave,linenums]
----
{
	"find": ("test"),
	"skip": 0,
	"batchSize": 10
}
----

The `executeCommand` returns a result like this:

[source,dataweave]
----
{
  "cursor": {
    "firstBatch": [
      {
        "_id": {
          "$oid": "..."
        },
        "meta": {
          "view": {
            "id": 1,
//            ...
           }
//         ,{ ... }, ...
        }
//....
      }
    ],
    "id": 9041685167132828464,
    "ns": "test.test"
  },
  "ok": 1.0
}
----

> To get additional results until there are no more results left to get, use the https://docs.mongodb.com/v4.4/reference/command/getMore/[`getMore` command] to invoke an `executeCommand` that retrieves the next page of the results:

Reviewer: ```suggestion
To get additional results until there are none left, use the https://docs.mongodb.com/v4.4/reference/command/getMore/[`getMore` command] to retrieve the next page of results:
```

[source,dataweave,linenums]
----
{
	"getMore": 9041685167132828464,
	"collection" : "test",
	"batchSize": 10
}
----

> The results contain the same structure with the cursor ID, which is used in subsequent command invocations for `getMore`.

Reviewer: ```suggestion
The results of the `getMore` command include a cursor ID to use in subsequent `getMore` commands.
```
